dagshub/models/model_locator.py

```diff
@@ -113,7 +109,7 @@ def _handle_path(self, path: str) -> Tuple[str, StorageType]:
         if str_path.startswith("dagshub_storage/"):
             return str_path[len("dagshub_storage/") :], StorageType.DagshubStorage
         for storage in self.repo_storages:
-            if str_path.startswith(f"{storage.name}/"):
+            if str_path.startswith(f"{storage.name}"):
```

Comment:
This is actually also wrong if something like this happens:
```python
str_path = a/b/c/def
storage.name = a/b/c/d
```
I think you need to make sure they both end with a slash and compare that

Reply:
Ok, I get the bug now, thanks for catching it
I don't think forcing a slash is a good idea (will break cases where path points to a yaml), so I'll check for either a prefix with a slash, or a full match